pkg/session/session.go

```diff
@@ -122,9 +127,17 @@ func GetOrCreate(ctx context.Context, params *Params) (*Session, error) {
 			logger.Error(err, "unable to check if rest session is active")
 		}
 
-		if vimSessionActive && tagManagerSession != nil {
-			logger.V(2).Info("found active cached vSphere client session")
-			return s, nil
+		if vimSessionActive {
+			returnCached := true
+			if tagManagerSession == nil && params.refreshRestClient {
```

Comment:
what will happen to s.TagManager.Session(ctx) if TagManager already logged out? will it return any error? Or shall we not do the check if refreshRestClient is false?

Reply:
We should get both session and error = nil

Reply:
sounds good. how about we don't do this check when refreshRestClient is false? because it's yet another API call to vcenter which is not needed.

Reply:
That makese sense let me chagne that